File: app/src/main/java/com/onedive/passwordstore/utils/Const.java
package com.onedive.passwordstore.utils;

/**
 * This class contains immutable values, which are used by this application
 * @author Ridh
 */
public final class Const {

    static {
        System.loadLibrary("native-lib");
    }

    // Base Url to check for updates
    public static final String UPDATE_BASE_URL = "https://api.jsonbin.io/v3/";

    //NAVIGATION KEY DATA THAT CONTAINS VALUES SENT TO OTHER ACTIVITIES

    // MainActivity / ListByTagActivity -> DetailPasswordActivity
    public static final String EXTRA_DETAIL_KEY = "ExtraDetail";

    //MainActivity -> ListByTagActivity
    public static final String EXTRA_LIST_BY_TAG = "ExtraTag";

    // MainActivity / DetailPasswordActivity  -> EditPasswordActivity
    public static final String EXTRA_EDIT = "ExtraEdit";


    //PREFERENCE key data is used to store application settings preferences
    public static final String LOCK_APP_KEY_PREFERENCE = "lockPreference";
    //public static final String ENCRYPT_KEY_PREFERENCE = "encryptPreference";

    //DEVELOPER CONTACT NUM
    public static final String DEVELOPER_CONTACT_NUMBER = "62895323021645";

    /**
     * Default encryption key, implemented using native c++,
     * you may be able to change the way the key is stored, one of them is using encrypt shared preference
     * @return default key in native c++
     */
    public static native String getNativeEncryptKey();

    /**
     *
     * @return JSON BIN ID in native c++
     */
    public static native String getUpdateJsonBinId();

    /**
     *
     * @return X-MASTER-KEY for JSON BIN in native c++
     */

    public static native String getApiSecretKey();

}

File: app/src/main/java/com/onedive/passwordstore/utils/Const.kt
package com.onedive.passwordstore.utils

/**
 * This class contains immutable values, which are used by this application
 * @author Ridh
 */
object Const {

    init { System.loadLibrary("native-lib") }

    // Base Url to check for updates
    const val UPDATE_BASE_URL = "https://api.jsonbin.io/v3/"


    // MainActivity / ListByTagActivity -> DetailPasswordActivity
    const val EXTRA_DETAIL_KEY = "ExtraDetail"

    //MainActivity -> ListByTagActivity
    const val EXTRA_LIST_BY_TAG = "ExtraTag"

    // MainActivity / DetailPasswordActivity  -> EditPasswordActivity
    const val EXTRA_EDIT = "ExtraEdit"



    const val LOCK_APP_KEY_PREFERENCE = "lockPreference"

    const val LOCK_DETAIL_KEY_PREFERENCE = "lockDetailPreference"


    //DEVELOPER CONTACT NUM
    const val DEVELOPER_CONTACT_NUMBER = "62895323021645"


    @Deprecated(
        message = "It is not safe to use this method because the encryption key stored in C++ " +
                "is not encrypted. So please use the encryption shared preference",
        replaceWith = ReplaceWith(expression = "encryptedSharedPreferences(context)"),
        level = DeprecationLevel.ERROR
    )
    /**
     * Default encryption key, implemented using native c++,
     * you may be able to change the way the key is stored, one of them is using encrypt shared preference
     * @return default key in native c++
     */
    external fun getNativeEncryptKey(): String

    /**
     *
     * @return JSON BIN ID in native c++
     */
    external fun getUpdateJsonBinId(): String

    /**
     *
     * @return X-MASTER-KEY for JSON BIN in native c++
     */
    external fun getApiSecretKey(): String

}